packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -92,23 +93,49 @@ import type { EventResult, PartialReference } from "@calcom/types/EventManager";
 
 import type { EventTypeInfo } from "../../webhooks/lib/sendPayload";
 import getBookingDataSchema from "./getBookingDataSchema";
+import type { BookingSeat } from "./handleSeats";
 
 const translator = short();
 const log = logger.getSubLogger({ prefix: ["[api] book:user"] });
 
 type User = Prisma.UserGetPayload<typeof userSelect>;
 type BufferedBusyTimes = BufferedBusyTime[];
 type BookingType = Prisma.PromiseReturnType<typeof getOriginalRescheduledBooking>;
-type Booking = Prisma.PromiseReturnType<typeof createBooking>;
+export type Booking = Prisma.PromiseReturnType<typeof createBooking>;
 export type NewBookingEventType =
   | Awaited<ReturnType<typeof getDefaultEvent>>
   | Awaited<ReturnType<typeof getEventTypesFromDB>>;
 
 // Work with Typescript to require reqBody.end
 type ReqBodyWithoutEnd = z.infer<ReturnType<typeof getBookingDataSchema>>;
 type ReqBodyWithEnd = ReqBodyWithoutEnd & { end: string };
-
-interface IEventTypePaymentCredentialType {
+export type Invitee = {
+  email: string;
+  name: string;
+  firstName: string;
+  lastName: string;
+  timeZone: string;
+  language: {
+    translate: TFunction;
+    locale: string;
+  };
+}[];
+export type OrganizerUser = Awaited<ReturnType<typeof loadUsers>>[number] & {
+  isFixed?: boolean;
+  metadata?: Prisma.JsonValue;
+};
+export type OriginalRescheduledBooking = Awaited<ReturnType<typeof getOriginalRescheduledBooking>>;
+export type RescheduleReason = Awaited<ReturnType<typeof getBookingData>>["rescheduleReason"];
+export type NoEmail = Awaited<ReturnType<typeof getBookingData>>["noEmail"];
+export type IsConfirmedByDefault = ReturnType<typeof getRequiresConfirmationFlags>["isConfirmedByDefault"];
+export type AdditionalNotes = Awaited<ReturnType<typeof getBookingData>>["notes"];
+export type ReqAppsStatus = Awaited<ReturnType<typeof getBookingData>>["appsStatus"];
+export type PaymentAppData = ReturnType<typeof getPaymentAppData>;
+export type SmsReminderNumber = Awaited<ReturnType<typeof getBookingData>>["smsReminderNumber"];
+export type EventTypeId = Awaited<ReturnType<typeof getBookingData>>["eventTypeId"];
```

Comment:
NIT

Lets save TS a little bit and so something like
```
type AwaitedBookingData = Awaited<ReturnType<typeof getBookingData>>`
export type RescheduleReason = AwaitedBookingData["rescheduleReason"];

Reply:
I love this. Thanks for the suggestion